### PR TITLE
[tune_gemm] Update the filter for LDS usage for stream-pipelineV2

### PR DESCRIPTION
**Pull request by zhanglx13**

### CODE REVIEW

File: python/perf-kernels/tools/tune_gemm/tune_gemm.py [hunk 1]
```diff
@@ -143,8 +143,14 @@ def prune_configs(M, N, K, configs, elemBytes_a, elemBytes_b):
             continue
         # out of shared memory resource
         # TODO (zhanglx): This does not consider the LDS usage in the epilogue
-        LDS = BLOCK_SIZE_K * BLOCK_SIZE_M * elemBytes_a + BLOCK_SIZE_K * BLOCK_SIZE_N * elemBytes_b
-        LDS = LDS if not num_stages else LDS * num_stages
+        LDSA = BLOCK_SIZE_K * BLOCK_SIZE_M * elemBytes_a
+        LDSB = BLOCK_SIZE_K * BLOCK_SIZE_N * elemBytes_b
+        if num_stages <= 1:
+            # No pipeline, buffer A and buffer B can re-use each other
+            LDS = min(LDSA, LDSB)
```
Comment (guacamoleo) [inline python/perf-kernels/tools/tune_gemm/tune_gemm.py]: I think we want max(LDSA, LDSB) to be able to store the larger.

**Review by guacamoleo — CHANGES_REQUESTED** [review]
Thanks for putting this together so quick!

**Comment by jtang10** [comment]
I'd suggest to still consider num_stages=0 for now, change it to 2 and add a warning for deprecation, considering some ppl might still be using older version.

**Comment by zhanglx13** [comment]
> I'd suggest to still consider num_stages=0 for now, change it to 2 and add a warning for deprecation, considering some ppl might still be using older version.

We have a warning here to make sure user will change num_stages = 0 to 2: https://github.com/triton-lang/triton/blob/9d6736a501d0499348d48d192b6260338ca19da0/third_party/amd/backend/compiler.py#L219

**Review by guacamoleo — APPROVED** [review]

**Pull request merged**